pkgs/applications/science/math/sage/env-locations.nix

```diff
@@ -35,7 +35,7 @@ writeTextFile rec {
     export GRAPHS_DATA_DIR='${graphs}/share/graphs'
     export ELLCURVE_DATA_DIR='${elliptic_curves}/share/ellcurves'
     export POLYTOPE_DATA_DIR='${polytopes_db}/share/reflexive_polytopes'
-    export GAP_ROOT_DIR='${gap}/share/gap/build-dir'
+    export GAP_ROOT_DIR='${gap}/share/gap'
```

Comment:
I am not sure in which sense this is the "GAP_ROOT_DIR", so the following may be totally off, but: The GAP root directory for building packages is `PREFIX/lib/gap`, as it contains `sysinfo.gap`. It is in `lib` and not `share` because thatfile contains architecture dependent information, in form of the `GAParch` entry.

Which of course in itself may be undesirable for you, as it might reveal data about the build machine?  One way to deal with that is to override it with a "generic" value, by adding `ARCH=something` to the `configure` arguments.

Reply:
That's a good point, I am also unsure about this. AFAICT [from the Sage source](https://github.com/sagemath/sage/blob/develop/src/sage/libs/gap/util.pyx#L247), `-l GAP_ROOT_DIR` is passed as part of argv to `GAP_Initialize`. Again AFAICT, that's the only use when Sage is using libgap, and I don't think this is used when Sage communicates with GAP via pexpect.

Reply:
Aha! OK, yeah the library files are in `share/gap`. So if that's what it is used for, then it is probably OK. Though the need for this could also be avoided by `#define`ing `SYS_DEFAULT_PATHS` suitable -- that's how the Fedora package does it. It would also be taken care of "automatically" with my proposal at https://github.com/gap-system/gap/issues/5096